prov/rxm/src/rxm_init.c

```diff
@@ -38,25 +38,22 @@
 int rxm_info_to_core(uint32_t version, const struct fi_info *hints,
 		     struct fi_info *core_info)
 {
-	/* Support modes that ofi_rxm could handle */
-	if (FI_VERSION_GE(version, FI_VERSION(1, 5)))
-		core_info->domain_attr->mr_mode |= FI_MR_LOCAL;
-	else
-		core_info->mode |= FI_LOCAL_MR;
+	core_info->mode |= FI_RX_CQ_DATA | FI_CONTEXT;
+	core_info->ep_attr->type = FI_EP_MSG;
 
-	/* The RxM FI_RMA implementation is pass-through but the provider
-	 * can handle FI_MR_PROV_KEY and FI_MR_VIRT_ADDR in its large message
-	 * transfer rendezvous protocol. */
-	if (!hints || !ofi_rma_target_allowed(hints->caps)) {
-		if (FI_VERSION_GE(version, FI_VERSION(1, 5)))
-			core_info->domain_attr->mr_mode |= OFI_MR_BASIC_MAP;
-		else
-			/* specify FI_MR_UNSPEC (instead of FI_MR_BASIC) so that
-			 * providers that support FI_MR_SCALABLE aren't dropped */
-			core_info->domain_attr->mr_mode = FI_MR_UNSPEC;
-	}
+	if (FI_VERSION_GE(version, FI_VERSION(1, 5))) {
+		core_info->domain_attr->mr_mode = OFI_MR_BASIC_MAP | FI_MR_LOCAL;
```

Comment:
We can't pass OFI_MR_BASIC_MAP mode bits to core provider in case app is a target of RMA. RxM doesn't handle exchanging of keys or address offsets in case of RMA. The app needs to be informed if the core provider requires FI_MR_BASIC.